src/twisted/protocols/_smb/security_blob.py

```diff
@@ -0,0 +1,329 @@
+# -*- test-case-name: twisted.protocols._smb.tests -*-
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+"""Implement the "security buffer" an intricate sub-protocol
+used for authentication
+
+Blobs implemented as DER-encoded ASN.1 objects
+which in turn encapsulate NTLMv2 challenge-response authentication
+
+
+much of code adapted from Mike Teo's pysmb """
+
+from typing import Optional, cast
+
+from pyasn1.codec.der import decoder, encoder  # type: ignore
+from pyasn1.type import char, constraint, namedtype, namedval, tag, univ  # type: ignore
+
+from twisted.logger import Logger
+from twisted.protocols._smb import _base, ntlm
+
+log = Logger()
+
+
+class BlobManager:
+    """
+    encapsulates the authentication negotiation state
+    callers just send blobs in and out and get a credential
```

Comment:
can you describe this briefly in terms of the order of the methods callers are expected to use? I don't know a ton about NTLM nd it's a little hard to imagine the expectations here.